src/main/java/com/plivo/api/models/base/BaseRequest.java

```diff
@@ -84,6 +94,18 @@ protected void handleResponse(Response response) throws PlivoRestException, IOEx
         throw new InvalidRequestException(response.errorBody().string());
       case 401:
         throw new AuthenticationException(response.errorBody().string());
+      case 403:
+        try {
+          String url = response.raw().request().url().toString();
+          if ((this instanceof Creator || this instanceof VoiceCreator) && 
+              GEO_PERMISSION_ENDPOINTS.stream().anyMatch(endpoint -> url.endsWith(endpoint))) {
+            throw new GeoPermissionException(response.errorBody().string());
+          }
+        } catch (GeoPermissionException e) {
+          throw e;
+        } catch (Exception e) {
+          // nop
```

Comment:
throw e?

Reply:
Yeah. Throw GeoPermissionException. Any other exception while extracting url, etc will be ignored